skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +22,79 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(TypeError):
+    pass
+
+
+class GrammaredSequenceMeta(ABCMeta, type):
+    def __new__(mcs, name, bases, dct):
+        cls = super(GrammaredSequenceMeta, mcs).__new__(mcs, name, bases, dct)
+
+        # Only perform validation on classes that aren't abstract.
+        if (type(cls.default_gap_char) is not abstractproperty and
```

Comment:
I actually already have it set up in a hierarchy. There's a second set of guards nested inside the first. Also I don't think defining the bools really makes it more readable (and believe me it doesn't take much of any excuse for me to throw in a temp for clarity). The raw Python already says pretty much exactly why it's doing: only process it's not abstract. "x_defined" seems ambiguous in this context.